clients/venice-push-job/src/test/java/com/linkedin/venice/hadoop/VenicePushJobTest.java

```diff
@@ -155,6 +139,107 @@ public void testVPJCheckInputUpdateSchema() {
+    VenicePushJob vpj = mock(VenicePushJob.class);
+    doCallRealMethod().when(vpj).validateAndSetRmdSchemas(any(), any());
+    vpj.validateAndSetRmdSchemas(mockClient, setting);
+  }
+
+  @Test
+  public void testValidateAndRetrieveRmdSchema() {
+    PushJobSetting setting = new PushJobSetting();
+    setting.rmdField = DEFAULT_RMD_FIELD_PROP;
+    setting.storeName = TEST_STORE;
+    setting.controllerRetries = 1;
+    setting.valueSchemaId = 1;
+    ControllerClient mockClient = mock(ControllerClient.class);
+
+    MultiSchemaResponse mockResponse = mock(MultiSchemaResponse.class);
+    MultiSchemaResponse.Schema rmdSchema = mock(MultiSchemaResponse.Schema.class);
+    when(rmdSchema.getRmdValueSchemaId()).thenReturn(1);
+    when(rmdSchema.getId()).thenReturn(1);
+    when(rmdSchema.getSchemaStr()).thenReturn(RMD_SCHEMA_STR);
+    when(mockResponse.getSchemas()).thenReturn(new MultiSchemaResponse.Schema[] { rmdSchema });
+    when(mockResponse.isError()).thenReturn(false);
+    when(mockClient.getAllReplicationMetadataSchemas(eq(TEST_STORE))).thenReturn(mockResponse);
+    VenicePushJob vpj = mock(VenicePushJob.class);
+    doCallRealMethod().when(vpj).validateAndSetRmdSchemas(any(), any());
+    vpj.validateAndSetRmdSchemas(mockClient, setting);
+    Assert.assertEquals(setting.replicationMetadataSchemaString, RMD_SCHEMA_STR);
+    Assert.assertEquals(setting.rmdSchemaId, 1);
+  }
+
+  @Test(expectedExceptions = VeniceException.class)
+  public void testValidateAndRetrieveRmdSchemaWithMultipleRmdSchemas() {
+    PushJobSetting setting = new PushJobSetting();
+    setting.rmdField = DEFAULT_RMD_FIELD_PROP;
+    setting.storeName = TEST_STORE;
+    setting.controllerRetries = 1;
+    setting.valueSchemaId = 1;
+    ControllerClient mockClient = mock(ControllerClient.class);
+
+    MultiSchemaResponse mockResponse = mock(MultiSchemaResponse.class);
+    MultiSchemaResponse.Schema rmdSchemaV1 = mock(MultiSchemaResponse.Schema.class);
+    when(rmdSchemaV1.getRmdValueSchemaId()).thenReturn(1);
+    when(rmdSchemaV1.getId()).thenReturn(1);
+    when(rmdSchemaV1.getSchemaStr()).thenReturn(RMD_SCHEMA_STR);
+
+    MultiSchemaResponse.Schema rmdSchemaV2 = mock(MultiSchemaResponse.Schema.class);
+    when(rmdSchemaV2.getRmdValueSchemaId()).thenReturn(1);
+    when(rmdSchemaV2.getId()).thenReturn(2);
+    when(rmdSchemaV2.getSchemaStr()).thenReturn(RMD_SCHEMA_STR_V2);
+
+    when(mockResponse.getSchemas()).thenReturn(new MultiSchemaResponse.Schema[] { rmdSchemaV1, rmdSchemaV2 });
+    when(mockResponse.isError()).thenReturn(false);
+    when(mockClient.getAllReplicationMetadataSchemas(eq(TEST_STORE))).thenReturn(mockResponse);
+    VenicePushJob vpj = mock(VenicePushJob.class);
+    doCallRealMethod().when(vpj).validateAndSetRmdSchemas(any(), any());
+    vpj.validateAndSetRmdSchemas(mockClient, setting);
+  }
+
+  @Test(expectedExceptions = VeniceException.class)
```

Comment:
I think `VeniceException` is too broad to expect without doing further validation on it's message